optional_plugins/robot/avocado_robot/__init__.py

```diff
@@ -0,0 +1,126 @@
+from robot import run
+from robot.parsing.model import TestData
+
+
+class RobotTest(test.SimpleTest):
+
+    """
+    Run a Robot command as a SIMPLE test.
+    """
+
+    def __init__(self,
+                 name,
+                 params=None,
+                 base_logdir=None,
+                 job=None,
+                 external_runner=None):
+        super(RobotTest, self).__init__(name, params, base_logdir, job)
+
+    @property
+    def filename(self):
+        """
+        Returns the path of the robot test suite.
+        """
+        return self.name.name.split(':')[0]
+
+    def test(self):
+        """
+        Create the Robot command and execute it.
+        """
+        suite_name, test_name = self.name.name.split(':')[1].split('.')
+        logfile = output.LoggingFile(logger=[self.log], level=logging.INFO)
+        run(self.filename,
+            suite=suite_name,
+            test=test_name,
+            outputdir=self.outputdir,
+            stdout=logfile)
+
+
+class RobotLoader(loader.TestLoader):
+    """
+    Robot loader class
+    """
+    name = "robot"
+
+    def __init__(self, args, extra_params):
+        super(RobotLoader, self).__init__(args, extra_params)
+
+    def discover(self, url, which_tests=False):
+        avocado_suite = []
+        robot_suite = self._find_tests(TestData(parent=None, source=url))
+        for item in robot_suite:
+            for robot_test in robot_suite[item]:
+                test_name = "%s:%s.%s" % (url, item, robot_test)
+                avocado_suite.append((RobotTest, {'name': test_name}))
+        return avocado_suite
+
+    def _find_tests(self, data, test_suite={}):
+        test_suite[data.name] = []
+        for test in data.testcase_table:
+            test_suite[data.name].append(test.name)
+        for child_data in data.children:
+            self._find_tests(child_data, test_suite)
+        return test_suite
+
+    @staticmethod
+    def get_type_label_mapping():
+        return {}
+
+    @staticmethod
+    def get_decorator_mapping():
+        return {}
+
+
+class RobotCLI(CLI):
+
+    """
+    Run Robot Framework tests
+    """
+
+    name = 'robot'
+    description = "Robot Framework options for 'run' subcommand"
+
+    def configure(self, parser):
+        run_subcommand_parser = parser.subcommands.choices.get('run', None)
+        if run_subcommand_parser is None:
+            return
+
+        msg = 'robot framework options'
+        self.remote_parser = run_subcommand_parser.add_argument_group(msg)
+        self.remote_parser.add_argument('--robot', action='store_true',
+                                        default=False,
+                                        help='Specify the robot test suite'
+                                        ' file or directory')
+        self.configured = True
+
+    def run(self, args):
+        if getattr(args, 'robot', None):
+            loader.loader.clear_plugins()
+            loader.loader.register_plugin(RobotLoader)
```

Comment:
^^ ad1) I could imagine `avocado run --loaders file:-robot foo` to temporary disable `robot` plugin when discovering tests...